doc/how-to/app/app_roles.rst

```diff
@@ -0,0 +1,476 @@
+.. _application_roles:
+
+Application roles
+=================
+
+An application role is a Lua module that implements specific functions or logic.
+You can enable or turn off a particular role for certain instances in a :ref:`configuration <configuration>` without restarting these instances.
+A role is run when a configuration is loaded or reloaded.
+
+Roles can be divided into the following groups:
+
+-   Roles provided by Tarantool.
+    For example, the ``config.storage`` role can be used to make a Tarantool replica set act as a :ref:`configuration storage <centralized_configuration_storage_set_up_tarantool>`.
+-   Roles provided by third-party Lua modules.
+    For example, the `CRUD <https://github.com/tarantool/crud>`__ module provides the ``roles.crud-storage`` and ``roles.crud-router`` roles that enable CRUD operations in a sharded cluster.
+-   Custom roles that are developed as a part of a cluster application.
+    For example, you can create a custom role to define a stored procedure or implement a supplementary service, such as an email notifier or a replicator.
+
+This section describes how to develop custom roles.
+To learn how to enable and configure roles, see :ref:`configuration_application_roles`.
+
+..  NOTE::
+
+    Don't confuse application roles with other role types:
+
+    -   A role is a container for privileges that can be granted to users. Learn more in :ref:`access_control_concepts_roles`.
+    -   A role of a replica set in regard to sharding. Learn more in :ref:`vshard_config_sharding_roles`.
```

Comment:
```suggestion
    -   A user role is a container for privileges that can be granted to users. Learn more in :ref:`access_control_concepts_roles`.
    -   A sharding role of a replica set in regard to sharding. Learn more in :ref:`vshard_config_sharding_roles`.
```
Just an idea, to decrease readers' confusion a bit

Reply:
> A user role

Sounds better but a role can also has a role :) So, its better to keep it as is.

> A sharding role of a replica set in regard to sharding.

`sharding` meets two times in this case and add some confusion.